### ISSUE TITLE
crash report

### ISSUE THREAD

**Issue by phraktle** [arsenetar/dupeguru#663]
Crash when trying to do a Pictures > Content search.

```
Application Identifier: com.hardcoded-software.dupeguru
Application Version: 4.0.3
Mac OS X Version: Version 10.15.4 (Build 19E287)

Traceback (most recent call last):
  File "build/dupeGuru.app/Contents/Resources/py/cocoa/inter.py", line 259, in pulse
  File "build/dupeGuru.app/Contents/Resources/py/hscommon/gui/progress_window.py", line 101, in pulse
  File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 323, in _job_error
  File "build/dupeGuru.app/Contents/Resources/py/hscommon/jobprogress/performer.py", line 43, in _async_run
  File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 780, in do
  File "build/dupeGuru.app/Contents/Resources/py/core/scanner.py", line 137, in get_dupe_groups
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/scanner.py", line 31, in _getmatches
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 167, in getmatches
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 65, in prepare_pictures
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/cache_shelve.py", line 129, in purge_outdated
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/cache_shelve.py", line 47, in __delitem__
  File "build/dupeGuru.app/Contents/Resources/py/shelve.py", line 128, in __delitem__
KeyError: b'id:37948'
```

**Comment by glubsy** [comment]
Please use the search feature before reporting bugs. This has been reported multiple times already and is supposedly fixed in version 4.0.4.

**Comment by SeaCelo** [comment]
I'm seeing this error in OS X after updating to latest 10.15.5. It was working fine before the update. 

> `Application Identifier: com.hardcoded-software.dupeguru
> Application Version: 4.0.3
> Mac OS X Version: Version 10.15.5 (Build 19F96)
> 
> Traceback (most recent call last):
>   File "build/dupeGuru.app/Contents/Resources/py/cocoa/inter.py", line 259, in pulse
>   File "build/dupeGuru.app/Contents/Resources/py/hscommon/gui/progress_window.py", line 101, in pulse
>   File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 323, in _job_error
>   File "build/dupeGuru.app/Contents/Resources/py/hscommon/jobprogress/performer.py", line 43, in _async_run
>   File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 780, in do
>   File "build/dupeGuru.app/Contents/Resources/py/core/scanner.py", line 137, in get_dupe_groups
>   File "build/dupeGuru.app/Contents/Resources/py/core/pe/scanner.py", line 31, in _getmatches
>   File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 167, in getmatches
>   File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 65, in prepare_pictures
>   File "build/dupeGuru.app/Contents/Resources/py/core/pe/cache_shelve.py", line 129, in purge_outdated
>   File "build/dupeGuru.app/Contents/Resources/py/core/pe/cache_shelve.py", line 47, in __delitem__
>   File "build/dupeGuru.app/Contents/Resources/py/shelve.py", line 128, in __delitem__
> KeyError: b'id:38308'`
